spec/lib/miq_automation_engine/miq_ae_object_spec.rb

```diff
@@ -81,19 +81,72 @@ def value_match(value, xml_value)
     expect(result["vms"].length).to eq(2)
   end
 
-  it "#process_args_as_attributes with an array containing invalid entries" do
-    vm2 = FactoryGirl.create(:vm_vmware)
-    result = @miq_obj.process_args_as_attributes({"Array::vms" => "VmOrTemplate::#{@vm.id},fred::12,,VmOrTemplate::#{vm2.id}"})
-    expect(result["vms"]).to be_kind_of(Array)
-    expect(result["vms"].length).to eq(2)
-  end
+  describe "#process_args_as_attributes" do
+    let(:result) { @miq_obj.process_args_as_attributes("Array::my_values" => my_values) }
+
+    context "with an array containing invalid entries" do
+      let(:my_values) { "VmOrTemplate::#{@vm.id},fred::12,,VmOrTemplate::#{FactoryGirl.create(:vm_vmware).id}" }
```

Comment:
@syncrou 
is the ,, testing nil values? But wouldn't fred::12 raise the exception before it processes the nil value